mungegithub/features/repo-updates.go

```diff
@@ -111,6 +139,23 @@ func (o *RepoInfo) walkFunc(path string, info os.FileInfo, err error) error {
 	return nil
 }
 
+
+// decodeAssignmentConfig will parse the yaml header if it exists and unmarshal it into an assignmentConfig.
+// If no yaml header is found, do nothing
+// Returns an error if the file cannot be read or the yaml header is found but cannot be unmarshalled
+var mdStructuredHeaderRegex = regexp.MustCompile("^---\n(.|\n)*\n---")
```

Comment:
Should we make the quantifier non-greedy? `^---\n(.|\n)*?\n---`. Could the markdown files contain a `---` further down in the file?